manager/integration/tests/common.py

```diff
@@ -3552,7 +3573,121 @@ def minio_delete_random_backup_block(client, core_api, volume_name):
 
 # TODO: implement nfs_delete_random_backup_block
 def nfs_delete_random_backup_block(client, core_api, volume_name):
-    pass
+    raise NotImplementedError
+
+
+def backupstore_count_backup_block_files(client, core_api, volume_name):
+    backup_target_setting = client.by_id_setting(SETTING_BACKUP_TARGET)
+    backupstore = backup_target_setting.value
+
+    if is_backupTarget_s3(backupstore):
+        return minio_count_backup_block_files(client, core_api, volume_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_count_backup_block_files()
+
+
+def nfs_count_backup_block_files():
+    raise NotImplementedError
+
+
+def minio_count_backup_block_files(client, core_api, volume_name):
+    backup_target_credential_setting = client.by_id_setting(
+            SETTING_BACKUP_TARGET_CREDENTIAL_SECRET)
+
+    secret_name = backup_target_credential_setting.value
+
+    minio_api = get_minio_api(client, core_api, secret_name)
+    bucket_name = get_backupstore_bucket_name(client)
+    prefix = minio_get_volume_backup_prefix(volume_name) + "/blocks"
+
+    block_object_files = minio_api.list_objects(bucket_name,
+                                                prefix=prefix,
+                                                recursive=True)
+    block_object_files_list = list(block_object_files)
+
+    return len(block_object_files_list)
+
+
+def backupstore_create_dummy_in_progress_backup(client, core_api, volume_name):
+    backup_target_setting = client.by_id_setting(SETTING_BACKUP_TARGET)
+    backupstore = backup_target_setting.value
+
+    if is_backupTarget_s3(backupstore):
+        return minio_create_dummy_in_progress_backup(client,
+                                                     core_api,
+                                                     volume_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_create_dummy_in_progress_backup()
+
+
+def nfs_create_dummy_in_progress_backup():
+    raise NotImplementedError
+
+
+def minio_create_dummy_in_progress_backup(client, core_api, volume_name):
+    backup_target_credential_setting = client.by_id_setting(
+            SETTING_BACKUP_TARGET_CREDENTIAL_SECRET)
+
+    secret_name = backup_target_credential_setting.value
+
+    minio_api = get_minio_api(client, core_api, secret_name)
+    bucket_name = get_backupstore_bucket_name(client)
+    prefix = minio_get_volume_backup_prefix(volume_name) + "/backups"
+
+    dummy_backup_cfg_data = {"Name": "dummy_backup",
+                             "VolumeName": volume_name,
+                             "CreatedTime": ""}
+
+    backup_cfg_file = "/tmp/backup_backup-dummy.cfg"
+
+    with open(backup_cfg_file, 'w') as bkp_cfg_file:
+        json.dump(dummy_backup_cfg_data, bkp_cfg_file)
+
+    try:
+        with open(backup_cfg_file, 'rb') as bkp_cfg_file:
+            bkp_cfg_file_stat = os.stat(backup_cfg_file)
+            minio_api.put_object(bucket_name,
```

Comment:
unless the put_object needs a file handle, you can use
[json.dumps](https://docs.python.org/3/library/json.html#json.dumps) to turn the python object into a json string directly.

Reply:
I've used `put_object` according to [minio's docs](https://docs.min.io/docs/python-client-api-reference.html#put_object)